enclave/src/rpc/mocks/rpc_gateway_mock.rs

```diff
@@ -52,20 +55,23 @@ impl RpcGatewayMock {
     }
 
     pub fn mock_place_order(order_uuid: Option<OrderUUID>, do_authorize: bool) -> Self {
+        create_in_memory_nonce_storage().unwrap();
         let mut get_place_order_mock = RpcGatewayMock::default();
         get_place_order_mock.order_uuid = order_uuid;
         get_place_order_mock.do_authorize = do_authorize;
         get_place_order_mock
     }
 
     pub fn mock_cancel_order(order_uuid: Option<OrderUUID>, do_authorize: bool) -> Self {
+        create_in_memory_nonce_storage().unwrap();
```

Comment:
With this, whenever you create a mock order within a unit test, you are also testing the nonce storage. So when this fails due to whatever reason, the whole unit test will fail, even though the actual order is actually working. Hence, you're not actually unit testing anymore.

Reply:
The mocks were replaced by fixed data.